GameOfLife/Calculations/NextCellGeneration.cs

```diff
@@ -0,0 +1,104 @@
+﻿namespace GameOfLife
+{
+    public class NextCellGeneration
+    {
+
+        /// <summary>
+        /// /  Calculates next generation of cells and
+        ///  updates game field acording to these 4 rules:
+        ///     Any live cell with fewer than two live neighbours dies, as if by underpopulation.
+        //      Any live cell with two or three live neighbours lives on to the next generation.
+        //      Any live cell with more than three live neighbours dies, as if by overpopulation.
+        //      Any dead cell with exactly three live neighbours becomes a live cell, as if by reproduction.
+        /// </summary>
+        /// <param name="gameField"></param>
+        public int CellCalculation(int[,] gameField)
+        {
+
+            int lenght = gameField.GetLength(0);
+            int width = gameField.GetLength(1);
+            int[,] nextGenerationOfCells = new int[lenght, width];
+            int aliveCells = 0;
+            for (int i = 0; i < lenght; i++)
+            {
+                for (int j = 0; j < width; j++)
+                {
+                    int liveNeighbourCells = CalculateAliveNeighbourCount(lenght, width, i, j, gameField);
+
+                    if (gameField[i, j] == 1 && (liveNeighbourCells == 3 || liveNeighbourCells == 2))
+                    {
+                        nextGenerationOfCells[i, j] = 1;
+                    }
+                    else if (gameField[i, j] == 0 && (liveNeighbourCells == 3))
+                    {
+                        nextGenerationOfCells[i, j] = 1;
+                    }
+
+                    if (gameField[i, j] == 1)
+                    {
+                        aliveCells++;
+                    }
+
+                }
+            }
+            CopyNextGenerationArrayToGameFieldArray(lenght, width, nextGenerationOfCells, gameField);
+            return aliveCells;
+
+        }
+
+        public int CalculateAliveNeighbourCount(int lenght, int width, int i, int j, int[,] gameField)
```

Comment:
If this method is not called from outside this class, it should most likely be a private one. Same for next method